lua/lspconfig/server_configurations/denols.lua

```diff
@@ -5,7 +5,7 @@ local function buf_cache(bufnr, client)
   local params = {}
   params['referrer'] = { uri = vim.uri_from_bufnr(bufnr) }
   params['uris'] = {}
-  client.request_sync('deno/cache', params)
+  client.request('deno/cache', params, function() end, bufnr)
```

Comment:
does there need handle error in callback when request failed?

Reply:
I don't think its needed, since if it fails, it will just print an error like:
```
LSP[denols][Warning] Module not found "https://deno.land/x/oak@0.1.0/mod.ts".
    at file:///home/mrcool/dev/deno/lab/tmp/a2Z/a.ts:1:16
```
I don't think we need more then that

Also I added another commit https://github.com/neovim/nvim-lspconfig/pull/2569/commits/4d1e30c1e38d4aa8597fd03abd9a27cbaee89fb0 that removes weird looking ansi escape sequences from the server messages

Reply:
ref https://github.com/mattn/vim-lsp-settings/blob/de060d969b405296a17bb9eea7c98b1bb1b15999/settings/deno.vim#L271

Reply:
I didn't find a way to get the result of the request, I tried 
```lua
  client.request('deno/cache', params, function(...)
    vim.print(...)
  end, bufnr)
```
but I don't see the result there,

Also I think its not important, like I said a warning is already printed

Reply:
Ok I see https://github.com/neovim/neovim/blob/bf0ac4f241cd22f53e026767ed5df89bf63bcf37/runtime/lua/vim/lsp.lua#L1430  what I did is supposed to show the result as a second argument but I get `true` instead

Reply:
this is sync ? callback param is`function(err, result, ctx) do err handle end`

Reply:
I used function(err,result,ctx) but this is what I get for those arguments
```
nil
true
{
  bufnr = 1,
  client_id = 1,
  method = "deno/cache",
  params = {
    referrer = {
      uri = "file:///home/mrcool/dev/deno/lab/tmp/Hse/a.ts"
    },
    uris = {}
  }
}
```

so I get result = true, instead of result = {} (empty object)

Reply:
I tried with request_sync and its the same thing I don't get an error 
![image](https://user-images.githubusercontent.com/22427111/233841362-205ebd05-3136-49bb-b738-5fe967d26e20.png)

Also note that deno lsp doesn't report the error back to the client 
```
[DEBUG][2023-04-23 13:56:22] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  id = 6222,  jsonrpc = "2.0"}
```

It just prints the warning I showed initially which I think is enough

Reply:
just assert(err) or other error handle notify or print. true mean success ? important is err :)

Reply:
I mean even in error case it will return true, because deno server doesn't report the error back

Reply:
your print is `nil, true ,{...}` is `err, result, ctx` . in callback we only want handle err if not error is success right ? why you care about result is true ?

Reply:
and this https://github.com/mattn/vim-lsp-settings/blob/de060d969b405296a17bb9eea7c98b1bb1b15999/settings/deno.vim#L271 check for error in a very old version of deno, I don't think this is relevant anymore

Reply:
I don't think this relate any version . if callback return an error why you don't handle it ?  and same as
```vim
 if a:data['response']['result'] == v:true
        " Cache file would be enabled after reload buffer.
        let l:curpos = getcurpos()
        execute 'edit ' . expand('%:p')
        call setpos('.', l:curpos)
        echo 'Cache file installed'
    endif
```

in callback handler 

```lua
  if err then
       handle error
       return
  end

  if result == true then
     ....
     print('Cache file installed')
  end

Reply:
I added https://github.com/neovim/nvim-lspconfig/pull/2569/commits/2d9a7f685bfc986c4f4c19bb61c65e43f7d5867a

(But what I'm arguing is that the error will never happen, deno server will always respond with result = true even if it can't cache a file)

Reply:
https://github.com/denoland/deno/blob/bdffcb409fd1e257db280ab73e07cc319711256c/cli/lsp/language_server.rs#L175-L255

Reply:
The only way to hit the error branches in the match arms , is if the client sends a malformed message (which we do not)

In the case where it can't cache the file it shows a warning and return true https://github.com/denoland/deno/blob/bdffcb409fd1e257db280ab73e07cc319711256c/cli/lsp/language_server.rs#L224

Anyhow I see your point

Reply:
actually nvm you're right it probably can fail, though for some reason I can't reproduce it in practice

Reply:
ah I think the code is hitting this branch https://github.com/denoland/deno/blob/bdffcb409fd1e257db280ab73e07cc319711256c/cli/lsp/language_server.rs#L242 it just shows a warning and doesn't early return with an error

Reply:
also if there has `params` has some error then we can know it from response err.  the error handle not only for cache failed.